code/ruby/index1.rb

```diff
@@ -1,36 +1,32 @@
 #!/usr/bin/env ruby
+# frozen_string_literal: true
 
 require 'xapian'
 require 'json'
 require_relative 'support'
 
-
 ### Start of example code.
 def index_csv(data_path, db_path)
   # puts "#{data_path} #{db_path}"
   db = Xapian::WritableDatabase.new(db_path, Xapian::DB_CREATE_OR_OPEN)
   term_generator = Xapian::TermGenerator.new
   term_generator.stemmer = Xapian::Stem.new('en')
   parse_csv_file(data_path).each do |row|
-    rec = row.to_h
     doc = Xapian::Document.new
     term_generator.document = doc
-    term_generator.index_text(row["TITLE"].to_s, 1, 'S')
-    term_generator.index_text(row["DESCRIPTION"].to_s, 1, 'XD')
-    term_generator.index_text(row["TITLE"].to_s)
+    term_generator.index_text(row['TITLE'].to_s, 1, 'S')
+    term_generator.index_text(row['DESCRIPTION'].to_s, 1, 'XD')
+    term_generator.index_text(row['TITLE'].to_s)
     term_generator.increase_termpos
-    term_generator.index_text(row["DESCRIPTION"].to_s)
+    term_generator.index_text(row['DESCRIPTION'].to_s)
     doc.data = row.to_h.to_json
-    idterm = "Q#{row["id_NUMBER"]}"
+    idterm = "Q#{row['id_NUMBER']}"
     doc.add_boolean_term(idterm)
     db.replace_document(idterm, doc)
   end
 end
 ### End of example code.
 
-if ARGV.length < 2
-  abort "Usage #{__FILE__} DATAPATH DBPATH"
-end
+abort "Usage #{__FILE__} DATAPATH DBPATH" if ARGV.length < 2
```

Comment:
I find it curious that a style tool should push towards this as it seems less clear in a case like this where the action is the exceptional case as it emphasises the action over the check, but here the `ARGV.length < 2` is the part I'd want the reader to focus on (`<action> if <condition>` seems reasonable when the action is the interesting part).

(To be clear, this is not a review comment saying to change this - really just a thought I had while skimming the changes.)